### PR TITLE
Make some more updates to the community guidelines

### PR DESCRIPTION
**Pull request by mxstbr**
- Rephrases some things
- Adds more information
- Documents release process on `npm`
- Notes CoC in Community guidelines

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -92,17 +96,29 @@ Now you should have the sandbox running on `localhost:3000`. The Sandbox support
...
 
 When you commit our pre-commit hook will run, which executes `lint-staged`. It will run the linter automatically and warn you, if the code you've written doesn't comply with our code style.
 
+## Release process
+
+[Core team members](./CORE_TEAM.md) have the responsibility of pushing new releases to npm. The release process is as follows:
+
+1. Install `np` by Sindre Sorhus with `npm i -g np`. (note: you only have to do this once) `np` makes sure that everything is correct and runs tests and a build before it releases the new version for you.
+2. Make sure you have the latest changes and are on the master branch: `git checkout master && git pull origin master`
+3. Create a new branch based on the the version number, for example `git checkout -b 3.4.1`
```
Comment (morajabi) [inline CONTRIBUTING.md]: I think we can link to a video on this, maybe https://egghead.io/lessons/git-use-git-flow-as-a-reliable-version-control-model

Comment (morajabi) [reply CONTRIBUTING.md]: Also, if you wanted maybe I can record a short screencast dedicated to this stack for Styled Components

Comment (mxstbr) [reply CONTRIBUTING.md]: I don't think that's necessary given only three people have access to publishing to npm at the moment :blush: I'd rather you spend the time improving the website or something!

Comment (morajabi) [reply CONTRIBUTING.md]: Never thought of that this way 😂

**Pull request merged**